File: leetcode/179. Largest Number.py
class Element:
    def __init__(self, num):
        self.num = str(num)
    
    def __lt__(self, other):
        return self.num + other.num >= other.num + self.num


class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        
        elements = [Element(num) for num in nums]
        elements.sort()
        res = ''.join([element.num for element in elements])
        
        return str(int(res))

File: leetcode/179. Largest Number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        std::sort(nums.begin(), nums.end(), [](int a, int b){
            std::string as = std::to_string(a);
            std::string bs = std::to_string(b);
            return as + bs > bs + as;
        });

        if (nums[0] == 0) {
            return "0";
        }

        std::string res;
        for (auto num : nums) {
            res.append(std::to_string(num));
        }
        return res;
    }
};
